### PR TITLE
Fix a gradient is None issue in max_pool2d_with_indices_backward

### PR DESCRIPTION
**Pull request by desertfire**
Summary: For https://github.com/pytorch/torchdynamo/issues/1200

### CODE REVIEW

File: torchinductor/lowering.py [hunk 1]
```diff
@@ -2173,12 +2173,14 @@ def max_pool2d_with_indices_backward(
             h // stride[0] - max(0, (h - kernel_size[0]) // stride[0])
             for h in range(kernel_size[0] * 2)
         ]
+        + [1]
```
Comment (voznesenskym) [inline torchinductor/lowering.py:2176]: Why does adding [1] here fix it?

Comment (desertfire) [reply torchinductor/lowering.py:2176]: It's to make h_window_size/w_window_size at least has value 1. I can rewrite this to use another `max` if that makes things more clear.

Comment (eellison) [inline torchinductor/lowering.py:2176]: nit, i think it would be a little more intuitive to fold the one into the expression than as `+ [1]`

**Review by eellison — APPROVED** [review]
LGTM - factor out / correct the window calculation fix for `avg_pool2d_backward` as well ?

**Review by jansel — APPROVED** [review]

**Pull request merged**